openverse_catalog/dags/providers/provider_api_scripts/cleveland_museum.py

```diff
@@ -1,97 +1,57 @@
 import logging
+from typing import Dict, Optional
 
 from common.licenses import get_license_info
 from common.loader import provider_details as prov
-from common.requester import DelayedRequester
-from common.storage.image import ImageStore
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
 
 
-LIMIT = 1000
-DELAY = 5.0
-RETRIES = 3
-PROVIDER = prov.CLEVELAND_DEFAULT_PROVIDER
-ENDPOINT = "http://openaccess-api.clevelandart.org/api/artworks/"
-
-delay_request = DelayedRequester(delay=DELAY)
-image_store = ImageStore(provider=PROVIDER)
-
-DEFAULT_QUERY_PARAMS = {"cc": "1", "has_image": "1", "limit": LIMIT, "skip": 0}
-
 logging.basicConfig(
     format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s", level=logging.INFO
 )
 logger = logging.getLogger(__name__)
 
 
-def main():
-    logger.info("Begin: Cleveland Museum API requests")
-    condition = True
-    offset = 0
-
-    while condition:
-        query_param = _build_query_param(offset)
-        response_json, total_images = _get_response(query_param)
-        if response_json is not None and total_images != 0:
-            batch = response_json["data"]
-            image_count = _handle_response(batch)
-            logger.info(f"Total images till now {image_count}")
-            offset += LIMIT
+class ClevelandDataIngester(ProviderDataIngester):
+    def __init__(
+        self,
+        providers={"image": prov.CLEVELAND_DEFAULT_PROVIDER},
+        endpoint="http://openaccess-api.clevelandart.org/api/artworks/",
+        delay=5,
+        batch_limit=1000,
+        retries=3,
```

Comment:
What do you think about having these fields as class level attributes rather than parameters for the `__init__` function? Having them here makes me think that we might have cases where we'd override this for a certain run of the Cleveland workflow, which I don't think will be the case. That also prevents us from having to call `super().__init__()` on all of the subclasses!

Reply:
I'm glad you brought this up, because I've actually gone back and forth on this a lot 😓  The fields were originally implemented as class level attributes, but I wasn't completely sold because I wanted to enforce that `providers` and `endpoint`, at least, be defined in the subclasses, and also because of the overriding of `batch_limit`. I did play around with abstract properties, but ultimately went with this structure because it's also consistent with [the way we've implemented the MediaStores](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/storage/audio.py#L37).

You also suggested adding a `__main__` block to [run the data ingester](https://github.com/WordPress/openverse-catalog/pull/555#discussion_r904174641). I wonder if an option might be to define constants for the Cleveland (or whatever provider) defaults within its file, and add a `__main__` block that initializes the Ingester using those defaults and runs ingestion. We could use that from the `provider_dag_factory` and avoid all the new code paths to deal with the difference between the class/function approach, and then the individual providers won't need to `super().__init__()` just to set new defaults 🤔

Reply:
I guess the downside would be a slight disparity in the timing reported for the `pull_data` task, which would include the time to initialize the Ingester class along with stores/DelayedRequester in addition to ingestion. The logic to[ push the Store output paths to Xcoms](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/provider_dag_factory.py#L115) also still needs to be different for the old type of script and new ones that subclass the DataIngester, and that'll be harder to key off of 🤔

Reply:
Oh, those are good points! I don't think that initialization time will contribute _too much_ to the overall time, but it's a good thing to be thinking about 🤔 

For your other points, I guess I'm not sure why having those as class attributes would affect the aspects you mention 😮 I like the idea of having `providers` and `endpoint` be abstract properties! `batch_limit` could still be set at the class level - I think even if we change that Variable while a run is happening, the next run will respect whatever the new limit is since Airflow re-imports everything on each DAG run step. And I was thinking the main block could look something like:

```python
if __name__ == "__main__":
    ingester = ClevelandDataIngester()
    ingester.ingest_records(DEFAULT_ARGS)
```

Or something similar. I might be missing initialization steps in that though 😅 and FWIW I don't think we would need to hook the `__main__` section up to publish XComs or interact with any of the rest of the workflow steps - that's strictly used for testing the ingester locally without having to run the DAG!

Reply:
For sure, the extra steps I suggested in `main` and using `main` instead of the class would only be useful in the scenario where we were still using instance variables -- I was thinking we could get a cleaner transition between the old way of invoking provider scripts and the new way, but realized we can't quite pull it off because of the XCom issue. It's definitely not useful outside of that!